frontend/packages/dev-console/src/components/helm/HelmReleaseListPage.tsx

```diff
@@ -26,16 +36,61 @@ export const HelmReleaseListPage: React.FC<HelmReleaseListPageProps> = (props) =
       selector: { owner: 'helm' },
     },
   ];
+  React.useEffect(() => {
+    let destroyed = false;
+    fetchHelmReleases(namespace)
+      .then((releases) => {
+        if (!destroyed) {
+          setHelmReleases(releases);
+        }
+      })
+      .catch((err) => {
+        if (!destroyed) {
+          setLoadError(err.message || 'Unable to load Helm Releases');
+        }
+      });
+    return () => {
+      destroyed = true;
+    };
+  }, [namespace]);
+
+  if (loadError) {
+    return <StatusBox loaded loadError={loadError} label="Helm Releases" />;
+  }
+
+  if (!helmReleases) {
+    return <LoadingBox />;
+  }
+
+  const helmImage = () => (
+    <img
+      className="odc-helm-release__empty-list__image"
+      src={getImageForIconClass('icon-helm')}
+      alt=""
+    />
+  );
+
+  const contents =
+    helmReleases.length === 0 ? (
+      <EmptyState variant={EmptyStateVariant.full}>
+        <p className="odc-helm-release__empty-list__title">No Helm Releases found</p>
+        <EmptyStateIcon variant="container" component={helmImage} />
+        <Link to={installURL}>Install a Helm Chart from the developer catalog</Link>
+      </EmptyState>
+    ) : (
+      <Firehose resources={resources}>
+        <HelmReleaseList namespace={namespace} />
```

Comment:
Why not send `helmReleases` as a prop to `HelmReleaseList` and use that to render the list? Right now the page is making multiple calls to the helm release list API which is not needed.

Reply:
The `HelmReleaseList` fetches the releases whenever the secrets change since the releases are not watchable. We could potentially use the first list of releases but would still need fetching to be done by the `HelmReleaseList`.

Reply:
Right, I guess we could do something like sending the `fetchCustomResources` callback from `HelmReleaseListPage`  to `CustomResourceList`  and change the callback based on changes in secret but that would be out of scope for this bug fix. I think it's okay to have an extra API call for now.